src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -375,6 +375,12 @@ def get(self, hostname: str) -> Dict:
 
     @raise_if_no_orchestrator([OrchFeature.HOST_LABEL_ADD, OrchFeature.HOST_LABEL_REMOVE])
     @handle_orchestrator_error('host')
+    @EndpointDoc('',
+                 parameters={
+                     'hostname': (str, 'Hostname'),
+                     'labels': ([str], 'Host Labels'),
+                 },
+                 responses={200: None, 204: None})
```

Comment:
I understand the point of adding explicit documentation here, and I also understand that it's not easy to get how the current 'implicit' documentation is generated, but given this implies adding redundancy (`hostname` are `labels` are already defined in the function signature), why not generating both the documentation and the type-checking in the existing infrastructure?

So my suggestion below goes for finding a general solution for both issues instead of just fixing this specific issue.

---
### Automatic Type Documentation

Specifically, for the docs generation (I have to admit that this code is pretty convoluted):
* The `EndpointDoc` decorator simply sets the controller metadata under the function `.doc_info` dict (`controllers/__init__.py`).
* At `controllers/docs.py` the `_gen_paths` methods simply scrapes all methods with that kind of `doc_info` metadata.

The change here would involve modifying `__init__.py`'s `_split_parameters()` method, so when it's called as `_split_parameters(parameters)` becomes `_split_parameters(parameters, func)`:
* If parameters is None, the it tries to auto-guess parameters from the function signature.
* For that, this method will have to accept the function object `func` as extra param.
* From that function object it'll be easy to infer the parameter types via [Python `inspect`](https://docs.python.org/3/library/inspect.html) module: `inspect.signature(func)`.

---
### Automatic Runtime Type checking
For the type-check, this shouldn't be done from the docs code (and it doesn't make sense too as it's unrelated to documentation). That should better go as a new Cherrypy tool, as it needs to happen at runtime:
* This new tool would recover the original function (taking as an example our AuthManagerTool, this could possibly be get from cherrypy.request.handler.callable, but due to nested decorators, this might not be the case)
* From that obtained function, we would get the same function signature as for the automatic documentation above, and perform the runtime type checking against the actual input arguments.

Reply:
@epuertat IMHO the cleanup you're suggesting can be done in a follow-up PR as the main issue here is to only allow a list of labels (not accept any other type like string) when calling that endpoint.

Reply:
AFAIK the Dashboard Team agreement (some time ago) was -for bugfixes- to provide the bugfix so they can be easily backported if necessary and do the refactoring in follow-up PRs.

Reply:
@alfonsomthd yes, you're totally right. My only fear is that we spend more time fixing the specific issues (symptoms of a design that needs to be improved) than solving the underlying cause of the issues. Probably this specific defect will also happen to most endpoints, as almost none of them perform any kind of runtime validation... So we are fixing this only because it's the only use case QA has tested so far...

Reply:
As discussed with @avanthakkar & @alfonsomthd let's get this one merged and continue with the runtime checker in this tracker: https://tracker.ceph.com/issues/46423